processor/resourceprocessor/resource_processor.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resourceprocessor
+
+import (
+	"context"
+
+	resourcepb "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+
+	"github.com/open-telemetry/opentelemetry-collector/consumer"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/processor"
+)
+
+type resourceTraceProcessor struct {
+	resource *resourcepb.Resource
+	next     consumer.TraceConsumer
+}
+
+func newResourceTraceProcessor(next consumer.TraceConsumer, cfg *Config) *resourceTraceProcessor {
+	resource := createResource(cfg)
+	return &resourceTraceProcessor{
+		next:     next,
+		resource: resource,
+	}
+}
+
+// ConsumeTraceData implements the TraceProcessor interface
+func (rtp *resourceTraceProcessor) ConsumeTraceData(ctx context.Context, td consumerdata.TraceData) error {
+	return rtp.next.ConsumeTraceData(ctx, consumerdata.TraceData{
+		Node:         td.Node,
+		Resource:     rtp.resource,
+		Spans:        td.Spans,
+		SourceFormat: td.SourceFormat,
+	})
+}
+
+// GetCapabilities returns the Capabilities assocciated with the resource processor.
+func (rtp *resourceTraceProcessor) GetCapabilities() processor.Capabilities {
+	return processor.Capabilities{MutatesConsumedData: true}
+}
+
+// Shutdown is invoked during service shutdown.
+func (*resourceTraceProcessor) Shutdown() error {
+	return nil
+}
+
+type resourceMetricProcessor struct {
+	resource *resourcepb.Resource
+	next     consumer.MetricsConsumer
+}
+
+func newResourceMetricProcessor(next consumer.MetricsConsumer, cfg *Config) *resourceMetricProcessor {
+	resource := createResource(cfg)
+	return &resourceMetricProcessor{
+		next:     next,
+		resource: resource,
+	}
+}
+
+// GetCapabilities returns the Capabilities assocciated with the resource processor.
+func (rmp *resourceMetricProcessor) GetCapabilities() processor.Capabilities {
+	return processor.Capabilities{MutatesConsumedData: true}
```

Comment:
nit: once configured I would expect this to be true, however, since this is checked only once it seems worth to check if `Type` and `Labels`, especially considering that you need a merge method to merge resources. That said this is an optimization to an "unusual" configuration, feel free to keep as it is if it becomes troublesome.

Reply:
Done